`prefer-string-replace-all`: Report all `String#replace()` when the pattern has `g` flag (#1965)
sindresorhus/eslint-plugin-unicorn: docs/rules/prefer-string-replace-all.md

<!-- Do not manually modify this header. Run: `npm run fix:eslint-docs` -->

> The [`String#replaceAll()`](https://github.com/tc39/proposal-string-replaceall) method is both faster and safer as you don't have to escape the regex if the string is not a literal.
> Even if regexp has to be used, `String#replaceAll()` has more clear intention.

Reviewer: ```suggestion
The [`String#replaceAll()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll) method is both faster and safer as you don't have to use a regex and remember to escape it if the string is not a literal. And when used with a regex, it makes the intent clearer.
```
